webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -82,26 +95,34 @@ export class ContactsEffects {
     );
   }, { dispatch: false });
 
-  private setTitle(selected) {
-    const routeSnapshot = this.routeSnapshotService.get();
-    const deceasedTitle = routeSnapshot?.data?.name === 'contacts.deceased'
-      ? this.translateService.instant('contact.deceased.title') : null;
-    const title = deceasedTitle || selected.type?.name_key || 'contact.profile';
-    this.globalActions.setTitle(this.translateService.instant(title));
+  private setTitle(contactId) {
+    return this
+      .verifySelectedContactNotChanged(contactId)
```

Comment:
I wonder if we can move this to the check to the main promise chain (to line 67) and keep this function cleaner and doing what it's called (sets the title).

Reply:
Done